component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateDataHolder.java

```diff
@@ -64,4 +68,44 @@ public HttpService getHttpService() {
     public void setHttpService(HttpService httpService) {
         this.httpService = httpService;
     }
-}
\ No newline at end of file
+
+    /**
+     * Set account lock service.
+     *
+     * @return account lock service
+     */
+    public AccountLockService getAccountLockService() {
+
+        return accountLockService;
```

Comment:
Lets throw an Runtime exception if this is null.  WDYT?

Reply:
Shall we do the same for other places as well.

Reply:
Fixed with ed2968b2c84d173790ff71b500a0268434981d01